### PR TITLE
feat: enable to retrieve lifecycleContext from api

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

This adds a function in the api to retrieve a lifecycleContext externally. 
This is needed so external extension can retrieve it and attach it when executing actions. 
The lifecycleContext is initialized in the provider-registry when a container/kube connection is registered.

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

this is part of the splitting of #1923 

### How to test this PR?

N/A

### CODE REVIEW

File: packages/extension-api/src/extension-api.d.ts [hunk 1]
```diff
@@ -423,6 +423,10 @@ declare module '@podman-desktop/api' {
     export const onDidUnregisterContainerConnection: Event<UnregisterContainerConnectionEvent>;
     export const onDidRegisterContainerConnection: Event<RegisterContainerConnectionEvent>;
     export function getContainerConnections(): ProviderContainerConnection[];
+    export function getMatchingProviderLifecycleContext(
+      providerId: string,
+      providerConnectionInfo: ProviderContainerConnectionInfo | ProviderKubernetesConnectionInfo,
+    ): LifecycleContextImpl;
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts]: I don't see how we can return a LifecycleContextImpl object
it's an internal implementation

File: packages/extension-api/src/extension-api.d.ts [hunk 2]
```diff
@@ -423,6 +423,10 @@ declare module '@podman-desktop/api' {
     export const onDidUnregisterContainerConnection: Event<UnregisterContainerConnectionEvent>;
     export const onDidRegisterContainerConnection: Event<RegisterContainerConnectionEvent>;
     export function getContainerConnections(): ProviderContainerConnection[];
+    export function getMatchingProviderLifecycleContext(
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts]:
```suggestion
    export function getProviderLifecycleContext(
```

probably matching is too internal 
WDYT ?

Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts]: in the jsdoc it does not say if there is an exception if there is no context ?

do we get an error or we should return undefined in that case ?

Comment (lstocchi) [reply packages/extension-api/src/extension-api.d.ts]: What do you mean by jsdoc? Adding a comment to the signature?

Currently it throws an error when there is no context.

Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts]: yes comments 👍

**Review by benoitf — APPROVED** [review]

**Pull request merged**